server.h

```diff
@@ -204,6 +209,20 @@ class CTFOServer {
             } else {
               card_entry.ctfo_percentage = 0.5;
             }
+          };
+
+          for (size_t i = 0; i < candidates.size() / 2 && (max_count ? (i < max_count) : true); ++i) {
+            const CID hot_cid = candidates[i];
+            response.hot_cards.resize(response.hot_cards.size() + 1);
+            ResponseCardEntry& hot_card_entry = response.hot_cards.back();
+            const Card& hot_card = data.Get(hot_cid);
+            FillResponseCardEntry(hot_card_entry, hot_card);
+
+            const CID recent_cid = candidates[candidates.size() / 2 + i];
+            response.recent_cards.resize(response.recent_cards.size() + 1);
+            ResponseCardEntry& recent_card_entry = response.recent_cards.back();
```

Comment:
Nit: Since you're doing a `push_back()` anyway, it's as fast and a bit more readable to just return from the lambda. In other words, `FillResponseCardEntry` -> `GenerateCardForFeed`.